node/src/ui_gateway/websocket_supervisor_factory.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019-2021, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::ui_gateway::websocket_supervisor::{WebSocketSupervisor, WebSocketSupervisorReal};
+use actix::Recipient;
+use masq_lib::ui_gateway::NodeFromUiMessage;
+
+pub trait WebsocketSupervisorFactory: Send {
+    fn make(
+        &self,
+        port: u16,
+        recipient: Recipient<NodeFromUiMessage>,
+    ) -> std::io::Result<Box<dyn WebSocketSupervisor>>;
+}
+
+pub struct WebsocketSupervisorFactoryReal;
+
+impl WebsocketSupervisorFactory for WebsocketSupervisorFactoryReal {
+    fn make(
+        &self,
+        port: u16,
+        recipient: Recipient<NodeFromUiMessage>,
+    ) -> std::io::Result<Box<dyn WebSocketSupervisor>> {
+        WebSocketSupervisorReal::new(port, recipient)
+            .map(|positive| positive as Box<dyn WebSocketSupervisor>)
+    }
+}
+
+#[cfg(test)]
+pub mod mock {
+    use crate::ui_gateway::websocket_supervisor::WebSocketSupervisor;
+    use crate::ui_gateway::websocket_supervisor_factory::WebsocketSupervisorFactory;
+    use actix::Recipient;
+    use masq_lib::ui_gateway::NodeFromUiMessage;
+    use std::cell::RefCell;
+
+    #[derive(Default)]
+    pub struct WebsocketSupervisorFactoryMock {
```

Comment:
I bet you'll find that you use WebsocketSupervisorFactoryMock in only one file. If so, this code should be in that file, not here.